pkg/operator/sync_test.go

```diff
@@ -118,3 +123,246 @@ func TestWaitForDeploymentRollout(t *testing.T) {
 		})
 	}
 }
+
+func TestSyncValidatingWebhooks(t *testing.T) {
+	defaultHook := mapiv1.NewValidatingWebhookConfiguration()
+	referenceHook := mapiv1.NewValidatingWebhookConfiguration()
+	referenceHook.SetName("test")
+
+	testCases := []struct {
+		testCase       string
+		exisingWebhook func() *admissionregistrationv1.ValidatingWebhookConfiguration
+		expectUpdate   bool
+	}{
+		{
```

Comment:
can we have a testCase for "If should not override `service.beta.openshift.io/inject-cabundle` if it's changed"?

Reply:
It's ok to override it here, as long we don't expect the hash of `.spec` to eventually match. With `library-go` implementation this test case will already be a part of the bundle.

Reply:
Aside from library go, I'm not following "is ok to override it"? The whole reason to move this to the operator is to make sure the `clientConfig.caBundle` is not overridden by the reconciliation loop so we need a unit to validate that.

Reply:
I assumed the reason was to take the management out of CVO, so it won't deadlock on the attempts to update the resource. Here (in this implementation) this change will occur only once, and DeepDerivative won't reconcile again after `caBundle` gets updated, as the certificate field on `required` resource is always empty.

We don't want to carry the task of populating the caBundle in our operator. What is set in this field is on another operator to decide. So it is equally fine - reset it to empty or preserve the field.

Library-go on the other hand ~should~ do the full resource state management, and achieve the full equality for patched resource with resource hash and generation matching expected values.

Reply:
the caBundle field is set the the service-ca-operator. Please include a test that validate we are not trampling that field on a reconciliation loop.

Reply:
>Here (in this implementation) this change will occur only once, and DeepDerivative won't reconcile again after caBundle gets updated, as the certificate field on required resource is always empty.

This statement is exactly what we want to cover in a test regardless the implementation details (using DeepDerivative or not) the outcome must be the same, i.e the spec.caBundle field is not overridden by the reconciliation loop.

Reply:
Makes sense, but what do you think about covering this with e2e tests? I can put a followup PR addressing this comment.